lib/discordrb/data/activity.rb

```diff
@@ -0,0 +1,260 @@
+      attr_reader :join
+
+      # @return [String, nil] secret for spectating
+      attr_reader :spectate
+
+      # @return [String, nil] secret for a specific instanced match
+      attr_reader :match
+
+      # @!visibility private
+      def initialize(data)
+        @join = data['join']
+        @spectate = data['spectate']
+        @match = data['match']
+      end
+    end
+
+    # Assets for rich presence images and hover text
+    class Assets
+      # @return [String, nil] the asset ID for the large image of this activity
+      attr_reader :large_image_id
+
+      # @return [String, nil] text displayed when hovering over the large iamge
+      attr_reader :large_text
+
+      # @return [String, nil] the asset ID for the small image of this activity
+      attr_reader :small_image_id
+
+      # @return [String, nil]
+      attr_reader :small_text
+
+      # @return [String, nil] the application ID for these assets.
+      attr_reader :application_id
+
+      # @!visibility private
+      def initialize(data, application_id)
+        @application_id = application_id
+        @large_image_id = data['large_image']
+        @large_text = data['large_text']
+        @small_image_id = data['small_image']
+        @small_text = data['small_text']
+      end
+
+      # Utility function to get an Asset's large image URL.
+      # @param format [String, nil] If `nil`, the URL will default to `webp`. You can otherwise specify one of `webp`, `jpg`, or `png`.
+      # @return [String] the URL to the large image asset.
+      def large_image_url(format = 'webp')
+        API.asset_url(@application_id, @large_image_id, format)
+      end
+
+      # Utility function to get an Asset's large image URL.
+      # @param format [String, nil] If `nil`, the URL will default to `webp`. You can otherwise specify one of `webp`, `jpg`, or `png`.
+      # @return [String] the URL to the small image asset.
+      def small_image_url(format = 'webp')
+        API.asset_url(@application_id, @small_image_id, format)
+      end
+    end
+
+    # Contains information about an activity's party
+    class Party
+      # @return [String, nil]
+      attr_reader :id
+
+      # @return [Integer, nil]
+      attr_reader :current_size
+
+      # @return [Integer, nil]
+      attr_reader :max_size
+
+      # @!visibility private
+      def initialize(data)
+        @id = data['id']
+        @current_size, @max_size = data['size']
+      end
+    end
+  end
+
+  # A collection of the user's activities.
+  class ActivitySet
+    include Enumerable
+
+    # @!visibility private
+    def initialize(activities = [])
+      @activities = activities
+    end
+
+    # @!visibility private
+    # Implement each for Enumerable
+    def each(&block)
+      @activities.each(&block)
+    end
+
+    # @return [Array<Activity>] all activities
+    def to_a
+      @activities
+    end
+
+    # @return [Array<Activity>, nil] all activities of type {Activity::GAME}
```

Comment:
These are no longer nilable.
```suggestion
    # @return [Array<Activity>] all activities of type {Activity::GAME}
```